packages/server/src/fhir/operations/agentbulkstatus.ts

```diff
@@ -0,0 +1,73 @@
+import { allOk, badRequest, isOk, serverError } from '@medplum/core';
+import { FhirRequest, FhirResponse } from '@medplum/fhir-router';
+import { Agent, Bundle, BundleEntry, OperationDefinition, OperationOutcome, Parameters } from '@medplum/fhirtypes';
+import { getAuthenticatedContext } from '../../context';
+import { agentStatusHandler } from './agentstatus';
+import { getAgentsForRequest } from './agentutils';
+
+export const operation: OperationDefinition = {
+  resourceType: 'OperationDefinition',
+  name: 'agent-bulk-status',
+  status: 'active',
+  kind: 'operation',
+  code: 'bulk-status',
+  experimental: true,
+  resource: ['Agent'],
+  system: false,
+  type: true,
+  instance: false,
+  parameter: [{ use: 'out', name: 'return', type: 'Bundle', min: 1, max: '1' }],
+};
+
+/**
+ * Handles HTTP requests for the Agent $status operation.
+ * First reads the agent and makes sure it is valid and the user has access to it.
+ * Then tries to get the agent status from Redis.
+ * Returns the agent status details as a Parameters resource.
+ *
+ * @param req - The FHIR request.
+ * @returns The FHIR response.
+ */
+export async function agentBulkStatusHandler(req: FhirRequest): Promise<FhirResponse> {
+  const { repo } = getAuthenticatedContext();
+  const agents = await getAgentsForRequest(req, repo);
+  if (!agents?.length) {
+    return [badRequest('No agent(s) for given query')];
+  }
+
+  const promises = agents.map((agent) => agentStatusHandler({ ...req, params: { id: agent.id as string } }));
```

Comment:
Low pri: there is opportunity for optimization here, as the `agentStatusHandler` basically just reads a value from Redis.  We could use Redis `mget` to make this a single Redis call.

That's probably premature optimization, so we could do that in the future.

Reply:
Yeah was thinking about this too, but my conclusion was yeah, maybe too early and don't want to potentially expose ourselves to diverging implementations yet